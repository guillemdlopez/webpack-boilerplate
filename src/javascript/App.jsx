import smiley from '../style/images/smiley.png';
import React from 'react';

const App = () => {
  return (
    <div className="container">
      <h1 style={{ textAlign: 'center' }}>Hello World!</h1>
      <div id="smiley">
        <img src={smiley} id="smiley-img" alt="smiley" />
      </div>
      <div className="time-label"></div>
    </div>
  );
};

export default App;
