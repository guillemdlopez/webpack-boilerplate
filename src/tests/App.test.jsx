import { shallow } from 'enzyme';
import App from '../javascript/App';

describe('Testing the <App /> component', () => {
  const wrapper = shallow(<App />);

  test('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should contain the greeting message Hello World!', () => {
    expect(wrapper.find('h1').exists()).toBe(true);
    expect(wrapper.find('h1').text().trim()).toBe('Hello World!');
  });
});
