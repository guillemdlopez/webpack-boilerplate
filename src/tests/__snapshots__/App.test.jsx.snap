// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the <App /> component should render correctly 1`] = `
<div
  className="container"
>
  <h1
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    Hello World!
  </h1>
  <img
    alt="smiley"
    id="smiley-img"
    src={null}
  />
</div>
`;
